File: src/Common/Models/TransactionEntry.java
package Common.Models;

import Common.Models.Base.Entity;

import java.math.BigDecimal;

public class TransactionEntry extends Entity {

    private Transaction transaction;
    private Account account;
    private BigDecimal amount;
    private BigDecimal rate;

    public Transaction getTransaction() {
        return this.transaction;
    }

    public void setTransaction(Transaction transaction) {
        this.transaction = transaction;
    }

    public Account getAccount() {
        return this.account;
    }
    public void setAccount(Account account) {
        this.account = account;
    }

    public BigDecimal getAmount() {
        return this.amount;
    }

    public void setAmount(BigDecimal amount) {
        this.amount = amount;
    }

    public BigDecimal getRate() {
        return this.rate;
    }

    public void setRate(BigDecimal rate) {
        this.rate = rate;
    }
}

File: backend/Business.Models/Entities/TransactionEntry.cs
using Business.Core.Entities;

namespace Business.Models.Entities;

public class TransactionEntry : BaseEntity
{
	public required Transaction Transaction { get; set; }
	public Guid TransactionId { get; set; }
	public required Account Account { get; set; }
	public Guid AccountId { get; set; }
	public decimal Amount { get; set; }
	public decimal Rate { get; set; }
}
